### PR TITLE
#11021 Add API documentation for property-view and selection-service

### PR DESCRIPTION
**Pull request by ndoschek**
#### What it does

- Add missing API documentation for interfaces and classes in the property-view package
- Add missing API documentation for the Theia selection-service

Resolves https://github.com/eclipse-theia/theia/issues/11021

Contributed on behalf of STMicroelectronics.

Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>

#### How to test

--

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/property-view/src/browser/property-view-widget-provider.ts [hunk 1]
```diff
@@ -20,6 +20,9 @@ import { PropertyDataService } from './property-data-service';
 import { PropertyViewContentWidget } from './property-view-content-widget';
 
 export const PropertyViewWidgetProvider = Symbol('PropertyViewWidgetProvider');
+/**
+ * The `PropertyViewWidgetProvider` should be implemented to provide a new content property view widget for the given selection.
```
Comment (vince-fugnitto) [inline packages/property-view/src/browser/property-view-widget-provider.ts:24]: I'm not sure about the wording here, especially:

> to provide a new content property view widget for the given selection.

```suggestion
 * The `PropertyViewWidgetProvider` should be implemented to provide additional content property view content for the given selection.
```

File: packages/core/src/common/selection-service.ts [hunk 2]
```diff
@@ -13,16 +13,21 @@
 //
 // SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
 // *****************************************************************************
-
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (vince-fugnitto) [inline packages/core/src/common/selection-service.ts:16]: minor: we should probably add a newline after.

File: packages/property-view/src/browser/property-view-widget-provider.ts [hunk 3]
```diff
@@ -58,8 +61,10 @@ export interface PropertyViewWidgetProvider {
     updateContentWidget(selection: Object | undefined): void;
 
 }
+
 /**
- * `DefaultPropertyViewWidgetProvider` should be extended to provide a new content property view widget for the given selection.
+ * The `DefaultPropertyViewWidgetProvider` is the default abstract implementation of the {@link PropertyViewWidgetProvider}
+ * and should be extended to provide a new content property view widget for the given selection.
```
Comment (vince-fugnitto) [inline packages/property-view/src/browser/property-view-widget-provider.ts:67]: Same as above I believe, I'm a bit confused by "a new content property view widget".

File: packages/property-view/src/browser/resource-property-view/resource-property-view-widget-provider.ts [hunk 4]
```diff
@@ -20,6 +20,11 @@ import { inject, injectable } from '@theia/core/shared/inversify';
 import { DefaultPropertyViewWidgetProvider } from '../property-view-widget-provider';
 import { ResourcePropertyViewTreeWidget } from './resource-property-view-tree-widget';
 
+/**
+ * The provider extends the {@link DefaultPropertyViewWidgetProvider}
```
Comment (vince-fugnitto) [inline packages/property-view/src/browser/resource-property-view/resource-property-view-widget-provider.ts:24]: I would omit the first line as it does not offer additional information that the class does not already provide with:

```ts
export class ResourcePropertyViewWidgetProvider extends DefaultPropertyViewWidgetProvider {
```

File: packages/property-view/src/browser/property-view-widget-provider.ts (hunk 1 above)
Comment (ndoschek) [reply packages/property-view/src/browser/property-view-widget-provider.ts]: I would maybe go with `provide a property view content widget for the given selection.`.
We use the wording of a property view content widget also elsewhere. An it describes best what this provider does: for the given selection the provider registered, it returns a content widget (that contains the property in an arbitrary form).
What do you think about that?

**Comment by ndoschek** [comment]
Thank you for your review @vince-fugnitto! I addressed your comments - could you have another look what you think about it? Thanks!

File: packages/property-view/src/browser/property-view-widget-provider.ts (hunk 1 above)
Comment (vince-fugnitto) [reply packages/property-view/src/browser/property-view-widget-provider.ts]: I'm fine with that, I assume the overall idea is that there is a single property-view container, and based on the selection we choose one of many different property-view content widget to be rendered within.

**Comment by ndoschek** [comment]
@vince-fugnitto I squashed my commits and rebased to master

**Pull request merged**